### PR TITLE
feat!: rename DrawerModal to Drawer

### PR DESCRIPTION
**Pull request by geotrev**
## Description

Renames `DrawerModal` to `Drawer` to match [website language](https://garden.zendesk.com/components/drawer).

## Detail

No discrete component code was changed.

## Checklist

- ~~:ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)~~
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- ~~:guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)~~
- ~~:wheelchair: tested for WCAG 2.1 AA accessibility compliance~~
- ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

File: packages/modals/README.md [hunk 1]
```diff
@@ -38,3 +40,59 @@ import { Button } from '@zendeskgarden/react-buttons';
...
+    <TooltipModal.Body>Some content</TooltipModal.Body>
+    <TooltipModal.Footer>
+      <TooltipModal.FooterItem>
+        <Button>Click</Button>
+      </TooltipModal.FooterItem>
+    </TooltipModal.Footer>
+    <TooltipModal.Close aria-label="Close" />
+  </TooltipModal>
+</ThemeProvider>;
+```
```
Comment (geotrev) [inline packages/modals/README.md:98]: Seemed worth including these. .

**Review by jzempel — CHANGES_REQUESTED** [review]
Please add `migration.md` details

File: docs/migration.md [hunk 2]
```diff
@@ -58,6 +58,7 @@ consider additional positioning prop support on a case-by-case basis.
 #### @zendeskgarden/react-modals
 
 - `TooltipModal`: removed `popperModifiers` prop (see [note](#breaking-changes))
+- `DrawerModal`: renamed to `Drawer`
```
Comment (jzempel) [inline docs/migration.md:61]: [nit] can we alphabetize component bullet lists under the package headings?

**Review by jzempel — APPROVED** [review]
Approved, with a potential [nit] modification. Feel free to merge when ready.

**Review by jzempel — APPROVED** [review]

**Pull request merged**